### ISSUE TITLE
Not an issue, but a note

### ISSUE THREAD

**Issue by LagLifeYT** [devos50/qemu#63]
The iPod Touch 3G drops the NOR chip, replacing it with a partition of NAND. Emulation (if at all) of this device will probably require a multi partition NAND image. 

Another interesting feature would be the abillity to restore using an iTunes backup, or an IPSW file, and the device would just boot. This would require GPU emulation, as well as USB to the point where it could send a file over. Although this is probably not something that can be implemented in the short term, considering how complex that would be. Would be cool though! It would also close off yet another route Apple could go about shutting this down, if the modified images are no longer provided, and you'd have to go and download an actual iOS image from Apple.

**Comment by devos50** [comment]
Yes, I read somewhere before that the IT3G indeed replaced the NOR with a NAND partition. I think that would make things slightly easier for emulation though as getting the NOR interactions up and running did require quite some time.

> Another interesting feature would be the abillity to restore using an iTunes backup, or an IPSW file, and the device would just boot. This would require GPU emulation, as well as USB to the point where it could send a file over.

That's a great idea! I'm not sure why this would require GPU emulation though. For sure we would have to look into the USB protocol, which would be very useful in general to get up and running.

**Comment by LagLifeYT** [comment]
It requires GPU emulation otherwise you’d have to modify the rootfs to disable the GPU after restoring.

Also, if the iPod Touch 4G ever gets emulated, I could restore using an iTunes backup (I used to have one of these)

I wonder if the iPhone 2G can be emulated by slightly changing some code, as it looks like it used the same CPU as the iPod Touch 1G (or at least one that's close)
